examples/with_winit/src/lib.rs

```diff
@@ -88,30 +92,540 @@ struct RenderState<'s> {
+                }
+                self.prior_position = Some(position);
+            }
+            WindowEvent::RedrawRequested => {
+                let _rendering_span = tracing::trace_span!("Actioning Requested Redraw").entered();
+                let encoding_span = tracing::trace_span!("Encoding scene").entered();
+
+                let Some(RenderState { surface, window }) = &self.state else {
+                    return;
+                };
+                let width = surface.config.width;
+                let height = surface.config.height;
+                let device_handle = &self.context.devices[surface.dev_id];
+                let snapshot = self.stats.snapshot();
+
+                // Allow looping forever
+                self.scene_ix = self.scene_ix.rem_euclid(self.scenes.len() as i32);
+                self.aa_config_ix = self.aa_config_ix.rem_euclid(AA_CONFIGS.len() as i32);
+
+                let example_scene = &mut self.scenes[self.scene_ix as usize];
+                if self.prev_scene_ix != self.scene_ix {
+                    self.transform = Affine::IDENTITY;
+                    self.prev_scene_ix = self.scene_ix;
+                    window.set_title(&format!("Vello demo - {}", example_scene.config.name));
+                }
+                self.fragment.reset();
+                let mut scene_params = SceneParams {
+                    time: self.start.elapsed().as_secs_f64(),
+                    text: &mut self.simple_text,
+                    images: &mut self.images,
+                    resolution: None,
+                    base_color: None,
+                    interactive: true,
+                    complexity: self.complexity,
+                };
+                example_scene
+                    .function
+                    .render(&mut self.fragment, &mut scene_params);
+
+                // If the user specifies a base color in the CLI we use that. Otherwise we use any
+                // color specified by the scene. The default is black.
+                let base_color = self
+                    .base_color
+                    .or(scene_params.base_color)
+                    .unwrap_or(Color::BLACK);
+                let antialiasing_method = AA_CONFIGS[self.aa_config_ix as usize];
+                let render_params = vello::RenderParams {
+                    base_color,
+                    width,
+                    height,
+                    antialiasing_method,
+                };
+                self.scene.reset();
+                if let Some(resolution) = scene_params.resolution {
+                    // Automatically scale the rendering to fill as much of the window as possible
+                    // TODO: Apply svg view_box, somehow
+                    let factor = Vec2::new(width as f64, height as f64);
+                    let scale_factor = (factor.x / resolution.x).min(factor.y / resolution.y);
+                    self.transform *= Affine::scale(scale_factor);
```

Comment:
Why are you modifying `self.transform` here?

Changing this to match the prior logic resolves the crashing issue for me.

Reply:
Thanks for catching this! Fixing this mistake works for me, too.